In [14]:
import numpy as np
import scipy.stats as stats

# Задача 1

In [22]:
# Данные о напряжении батареек
battery = np.array([12.9, 11.6, 13.5, 13.9, 12.1, 11.9, 13.0])

# Вычисление t-критерия для одной выборки
# Гипотеза: среднее напряжение батареек равно 12 В (номинал)
t_stat, p_value = stats.ttest_1samp(battery, 12)

print("t =", t_stat, "p =", p_value)

t = 2.1578017974978794 p = 0.07430190210562303


Вывод: проведённый t-критерий для одной выборки позволяет проверить гипотезу о том, что среднее напряжение батареек равно 12 В

- Нулевая гипотеза (H_0): среднее напряжение батареек равно 12 В
- Альтернативная гипотеза (H₁): Среднее напряжение батареек не равно 12 В
- При уровне значимости альфа равное 0.05 полученное p-значение (0.0743) больше альфа

Заключение: нет достаточных оснований для отклонения нулевой гипотезы. Среднее напряжение батареек статистически не отличается от 12 В на уровне значимости 0.05

# Задача 2

In [ ]:

# Сравнение двух методов определения содержания жира
data = {
    'Метод 1': [23.1, 23.2, 26.5, 26.6, 27.1, 48.3, 40.5, 25.0, 38.4, 23.5],
    'Метод 2': [22.7, 23.6, 27.1, 27.4, 27.4, 46.8, 40.4, 24.9, 38.1, 23.8]
}
df = pd.DataFrame(data)

# Вычисление t-критерия для зависимых
# Гипотеза: оба метода дают одинаковые результаты
t_stat, p_value = stats.ttest_rel(df['Метод 1'], df['Метод 2'])

print("t =", t_stat, "p =", p_value)

t = 1.7154884978678027e-15 p = 0.9999999999999987


Вывод: использован t-критерий для зависимых выборок с целью проверить, отличаются ли средние значения результатов двух методов

- Нулевая гипотеза (H_0): cредние значения, полученные методами 1 и 2, равны
- Альтернативная гипотеза (H_1): cредние значения отличаются
- Полученное p-значение ( около 1 ) значительно превышает уровень значимости альфа равное 0.05

Заключение: нет оснований отклонять нулевую гипотезу. Нет статистически значимой разницы между методами 1 и 2 в определении содержания жира

# Задача 3

In [23]:
# Проверка гипотезы о разнице давления в городах и деревнях
city = np.array([132, 111, 119, 138, 200, 131, 138, 170, 159, 140])
village = np.array([115, 190, 127, 155, 148, 121, 116, 121, 197])

# Вычисление t-критерия для независимых выборок
# Гипотеза: давление в городах выше, чем в деревнях
t_stat, p_value = stats.ttest_ind(city, village)

print("t =", t_stat, "p =", p_value)

t = 0.035165199049777565 p = 0.972357624400922


Вывод: применён t-критерий для независимых выборок с целью проверить, отличаются ли средние значения артериального давления между горожанами и сельскими жителями

- Нулевая гипотеза (H_0): Среднее артериальное давление горожан равно среднему артериальному давлению сельских жителей
- Альтернативная гипотеза (H_1): Среднее артериальное давление горожан выше, чем у сельских жителей
- Полученное p-значение (околко 0.972) значительно превышает уровень значимости альфа равное 0.05

Заключение: нет оснований отклонять нулевую гипотезу. Нет статистически значимой разницы в среднем артериальном давлении между жителями городов и деревень на уровне значимости 0.05

# Задача 4

In [24]:
# Проверка равенства вероятностей рождения мужчин и женщин
n = 5000  # Объем выборки
n_men = 2500  # Количество мужчин
p_men = 0.52  # Известная вероятность рождения мужчин

# Вычисление статистики z
p_hat = n_men / n
se = np.sqrt(p_men * (1 - p_men) / n)
z_stat = (p_hat - p_men) / se
p_value = stats.norm.sf(abs(z_stat)) * 2

print("z =", z_stat, "p =", p_value)

z = -2.830692585361492 p = 0.004644733942292467


Вывод: использован z критерий для проверки пропорций с целью определить, отличается ли доля мужчин в выборке от ожидаемой пропорции при равной смертности

- Нулевая гипотеза (H_0): доля мужчин в выборке равна 52% (смертность одинакова)
- Альтернативная гипотеза (H_1): Доля мужчин в выборке отличается от 52% (смертность различна)
- Полученное p-значение (около 0.004) меньше уровня значимости альфа равное 0.05

Заключение: есть достаточные основания отклонить нулевую гипотезу. На уровне значимости 0.05 можно считать, что смертность среди мужчин и женщин различается

# Задача 5

### Первый эксперимент:
- **Средние значения**: X1 = 17, X2 = 16
- **p-значение**: 0.001 (различия значимы)

### Второй эксперимент:
- **Средние значения**: X1 = 17, X2 = 36
- **p-значение**: 0.8 (различия не значимы)

## Возможные причины различий в результатах

### 1. Различие в вариабельности данных
- **Первый эксперимент**:
  - Низкая дисперсия внутри групп
  - Небольшая стандартная ошибка разности средних
  - Даже небольшая разница оказалась статистически значимой

- **Второй эксперимент**:
  - Высокая дисперсия внутри групп
  - Большая стандартная ошибка разности средних
  - Несмотря на большую разницу средних, p-значение высокое, и различия не значимы

### 2. Размер выборки
- **Малый размер выборки** во втором эксперименте мог привести:
  - Увеличению стандартной ошибки
  - Снижению статистической мощности теста
  - Высокому p-значению, несмотря на большую разницу средних

### 3. Качество данных и наличие выбросов
- Во втором эксперименте возможны:
  - Выбросы или аномалии
  - Неравномерное распределение данных
- Это увеличивает вариабельность и влияет на p-значение

### 4. Разные типы t-тестов
- Возможное использование разных вариантов t-теста:
  - С предположением о равных дисперсиях
  - Без предположения о равных дисперсиях
- Это может повлиять на результаты теста

## Вывод 
Наиболее вероятной причиной того, что во втором эксперименте большая разница средних оказалась незначимой, является **высокая вариабельность данных** или **малый размер выборки**.
Эти факторы увеличивают стандартную ошибку разности средних, что приводит к высокому p-значению
Несмотря на большую разницу средних во втором эксперименте, **высокая вариабельность данных** или **малый размер выборки** сделали различия статистически незначимыми